components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/KeyStoreAdmin.java

```diff
@@ -727,182 +726,126 @@ private static CertData[] doFilter(String filter, CertData[] certDataSet) {
             PaginatedKeyStoreData keyStoreData = fillPaginatedKeyStoreData(keyStoreName, keyStoreType,
                     paginatedCerts, paginatedKeyCerts);
 
             return keyStoreData;
         } catch (Exception e) {
-            String msg = "Error has encounted while loading the keystore to the given keystore name "
-                    + keyStoreName;
-            log.error(msg, e);
-            throw new SecurityConfigException(msg);
+            throw new SecurityConfigException(e.getMessage());
         }
     }
+
     /**
-     * Load the default trust store (allowed only for super tenant).
-     *
-     * @return trust store object
-     * @throws SecurityConfigException if retrieving the truststore fails.
+     * @param tenantId     Tenant Id.
+     * @param keyStoreName Keystore Name.
+     * @return
+     * @throws Exception
      */
-    public KeyStore getTrustStore() throws SecurityConfigException {
-        //Allow only the super tenant to access the default trust store.
-        if (tenantId != MultitenantConstants.SUPER_TENANT_ID) {
-            throw new SecurityConfigException("Permission denied for accessing trust store");
-        }
-
-        KeyStore trustStore;
-        ServerConfiguration serverConfiguration = ServerConfiguration.getInstance();
-        String file = new File(serverConfiguration.getFirstProperty(SERVER_TRUSTSTORE_FILE)).getAbsolutePath();
-
-        KeyStore store;
-        try {
-            store = KeyStore.getInstance(serverConfiguration.getFirstProperty(SERVER_TRUSTSTORE_TYPE));
-        } catch (KeyStoreException e) {
-            throw new SecurityConfigException("Error occurred while loading keystore.", e);
+    private KeyStore getKeyStore(int tenantId, String keyStoreName) throws Exception {
+        KeyStore keyStore;
+        if (KeyStoreUtil.isPrimaryStore(keyStoreName)) {
+            KeyStoreManager keyStoreManager = KeyStoreManager.getInstance(tenantId);
+            keyStore = keyStoreManager.getPrimaryKeyStore();
+        } else if (isTrustStore(keyStoreName)) {
+            keyStore = getTrustStore();
+        } else {
+            keyStore = getKeyStore(keyStoreName);
         }
+        return keyStore;
+    }
 
-        String password = serverConfiguration.getFirstProperty(SERVER_TRUSTSTORE_PASSWORD);
-
-        try (FileInputStream in = new FileInputStream(file)) {
-            store.load(in, password.toCharArray());
-            trustStore = store;
-        } catch (CertificateException | NoSuchAlgorithmException | IOException e) {
-            throw new SecurityConfigException("Error occurred while loading trust store", e);
+    /**
+     * Get keystore type.
+     *
+     * @param keyStoreName Keystore name.
+     * @return
+     * @throws SecurityConfigException
+     * @throws RegistryException
+     */
+    private String getKeyStoreType(String keyStoreName) throws SecurityConfigException, RegistryException {
+        String keyStoreType;
```

Comment:
Let's add newline after method signature. Fix in similar places.

Reply:
Fixed in all places.